common/djangoapps/third_party_auth/utils.py

```diff
@@ -30,6 +30,17 @@ def user_exists(details):
     return False
 
 
+def convert_slug_provider_id(provider):
+    # Provider id is stored with the backend type prefixed to it (ie "saml" or "oauth")
+    # Slug is stored without this prefix.
+    # This is only written to support SAML.
+    # This just converts between them whenever you expect the opposite of what you currently have.
+    if provider.startswith('saml-'):
```

Comment:
Up to you but suggested: Call me paranoid but shall we use the prefix field value from the SAMLProviderConfig, if it is even possible? or turn it up into a constant or something? Just thinking if someone has the crazy idea of changing that prefix value someday ... 

https://github.com/edx/edx-platform/blob/6d365ca1da62fbcf2ac442c258e3aca718214bce/common/djangoapps/third_party_auth/models.py#L553

but otherwise this looks like this will do it

Reply:
What would this look like? I'm not passing in a ProviderConfig object here and would like to keep it lightweight enough to not have one, but since prefix is a static field I could see just calling that directly on the general existence of the providerconfig model if that's a thing you can do (I'm honestly not sure)

The other issue is that we have plenty of slugs (in testing & stage anyways) that are prefixed with saml, so doing something like `if provider.startswith(prefix)` will catch both provider_ids and slugs. We specifically need to look for `saml-`

Reply:
yup you are aright, I __Was__ thinking of a static field constant that is exported from the providerConfig class but given the other thread of conversation this does not concern me as much anymore. My main concern was if someone changes the prefix in the model, but it seems lik ethat will need __a lot of__ changes anyway and I can't imaging why anyone would want to do that, so this is not concerning anymore for me

Reply:
yeah fair evaluation there.